cl_manycore/libraries/bsg_manycore_tile.cpp

```diff
@@ -81,102 +82,408 @@ int hb_mc_tile_set_origin(hb_mc_manycore_t *mc, const hb_mc_coordinate_t *tile,
+* */
+int hb_mc_tile_set_grid_dim_symbols (hb_mc_manycore_t *mc, hb_mc_eva_map_t *map, unsigned char* bin, size_t bin_size, const hb_mc_coordinate_t *coord, const hb_mc_dimension_t *grid_dim){
+
+	int error;
+	const hb_mc_config_t *cfg = hb_mc_manycore_get_config (mc); 
+
+	hb_mc_eva_t grid_dim_x_eva, grid_dim_y_eva;
+	error = hb_mc_loader_symbol_to_eva(bin, bin_size, "__bsg_grid_dim_x", &grid_dim_x_eva); 
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to aquire __bsg_grid_dim_x eva.\n", __func__);
+		return HB_MC_NOTFOUND;
+	}
+
+	error = hb_mc_loader_symbol_to_eva(bin, bin_size, "__bsg_grid_dim_y", &grid_dim_y_eva); 
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to aquire __bsg_grid_dim_y eva.\n", __func__);
+		return HB_MC_NOTFOUND;
+	}
+
+
+	hb_mc_npa_t grid_dim_x_npa, grid_dim_y_npa;
+	size_t sz;
+	error = hb_mc_eva_to_npa(cfg, map, coord, &grid_dim_x_eva, &grid_dim_x_npa, &sz);
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to aquire __bsg_grid_dim_x npa from eva.\n", __func__);
+		return HB_MC_FAIL;
+	}
+
+	error = hb_mc_eva_to_npa(cfg, map, coord, &grid_dim_y_eva, &grid_dim_y_npa, &sz);
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to aquire __bsg_grid_dim_y npa from eva.\n", __func__);
+		return HB_MC_FAIL;
+	}
+
+
+	error = hb_mc_manycore_write32(mc, &grid_dim_x_npa, hb_mc_dimension_get_x(*grid_dim));
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to write __bsg_grid_dim_x to tile (%d,%d).\n", __func__, hb_mc_coordinate_get_x(*coord), hb_mc_coordinate_get_y(*coord)); 
+		return HB_MC_FAIL;
+	}
+	bsg_pr_dbg("%s: Setting tile (%d,%d) __bsg_gird_dim_x (epa 0x%08" PRIx32 ") to %d.\n", __func__, hb_mc_coordinate_get_x(*coord), hb_mc_coordinate_get_y(*coord), grid_dim_x_npa.epa, hb_mc_dimension_get_x(*grid_dim));
+
+
+	error = hb_mc_manycore_write32(mc, &grid_dim_y_npa, hb_mc_dimension_get_y(*grid_dim));
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to write __bsg_grid_dim_y to tile (%d,%d).\n", __func__, hb_mc_coordinate_get_x(*coord), hb_mc_coordinate_get_y(*coord)); 
+		return HB_MC_FAIL;
+	}
+	bsg_pr_dbg("%s: Setting tile (%d,%d) __bsg_grid_dim_y (epa 0x%08" PRIx32 ") to %d.\n", __func__, hb_mc_coordinate_get_x(*coord), hb_mc_coordinate_get_y(*coord), grid_dim_y_npa.epa, hb_mc_dimension_get_y(*grid_dim));
+
+	return HB_MC_SUCCESS;
+}
+
+
+
+
+/*! 
+ * Writes a uint8_t into a tile given its coordinates and epa 
+ * Behavior is undefined if #mc is not initialized with hb_mc_manycore_init().
+ * @param[in] mc         A manycore instance initialized with hb_mc_manycore_init().
+ * @param[in] coord      Cooridnates of the destination tile
+ * @param[in] epa        Epa address in desitnation tile to which the write occurs
+ * @param[in] v          Variable to be written into tile 
+ * @return HB_MC_SUCCESS on success and HB_MC_FAIL on failure.
+* */
+int hb_mc_tile_write8(hb_mc_manycore_t *mc, const hb_mc_coordinate_t *coord, const hb_mc_epa_t *epa, uint8_t v) { 
```

Comment:
Can you use the hb_mc_manycore_write API directly instead of creating a new API?

Reply:
Done!